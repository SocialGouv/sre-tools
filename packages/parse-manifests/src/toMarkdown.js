import parseManifests from "./index.js";
import {
  getGrafanaLogsUrl,
  getGrafanaWorkloadsUrl,
  getRancherUrl,
  getGrafanaPodsUrl,
} from "./tools.js";

const toMarkdown = (manifests) => {
  const parsed = parseManifests(manifests);

  return `
<details>
  <summary>Ingresses</summary>

${parsed.hosts.map((host) => ` - 🚀 [https://${host}](${host})`).join("\n")}

</details>

<details>
  <summary>Docker images</summary>

${parsed.images.map((image) => ` - 📦 docker pull ${image}`).join("\n")}

</details>

<details>
  <summary>Debug</summary>

 - [📕 Loki logs for namespace ${parsed.namespace}](${getGrafanaLogsUrl(
    parsed
  )})
 - [📈 Pods monitoring for namespace ${parsed.namespace}](${getGrafanaPodsUrl(
    parsed
  )})
 - [📈 Workloads monitoring for namespace ${
   parsed.namespace
 }](${getGrafanaWorkloadsUrl(parsed)})
 - [👮‍♂️ Rancher project ${parsed.namespace}](${getRancherUrl(parsed)})

   </summary>
</details>
`;
};

export default toMarkdown;
