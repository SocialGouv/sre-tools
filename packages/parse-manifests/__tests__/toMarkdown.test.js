import toMarkdown from "../src/toMarkdown.js";

import fs from "fs";
import path from "path";

//console.log(parseManifests(fs.readFileSync("../sample.yml").toString()));

test("should render markdown", () => {
  const yaml = fs.readFileSync(path.join("sample.yml")).toString();
  expect(toMarkdown(yaml)).toMatchSnapshot();
});
