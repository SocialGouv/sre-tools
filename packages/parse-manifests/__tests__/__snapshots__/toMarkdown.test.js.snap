// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render markdown 1`] = `
"
<details>
  <summary>Ingresses</summary>

 - 🚀 [https://backend-app-preprod.dev.company.com/](https://backend-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/](https://version1-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/something](https://version1-app-preprod.dev.company.com/something)
 - 🚀 [https://version2-app-preprod.dev.company.com/](https://version2-app-preprod.dev.company.com/)

</details>

<details>
  <summary>Redirects</summary>

 - https://www.some.url ➡️ https://target.fr$request_uri
 - https://www.some.alternative.url ➡️ https://target.fr$request_uri

</details>

<details>
  <summary>Docker images</summary>

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

</details>

<details>
  <summary>Debug</summary>

 - [📕 Loki logs for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D)
 - [📈 Pods monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/dRAC0MRIz/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod)
 - [📈 Workloads monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/m_UCAGgIz/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment)
 - [🐘 CNPG some-cnpg-cluster](https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All)
</details>
"
`;

exports[`should render markdown with rancher details 1`] = `
"
<details>
  <summary>Ingresses</summary>

 - 🚀 [https://backend-app-preprod.dev.company.com/](https://backend-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/](https://version1-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/something](https://version1-app-preprod.dev.company.com/something)
 - 🚀 [https://version2-app-preprod.dev.company.com/](https://version2-app-preprod.dev.company.com/)

</details>

<details>
  <summary>Redirects</summary>

 - https://www.some.url ➡️ https://target.fr$request_uri
 - https://www.some.alternative.url ➡️ https://target.fr$request_uri

</details>

<details>
  <summary>Docker images</summary>

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

</details>

<details>
  <summary>Debug</summary>

 - [📕 Loki logs for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D)
 - [📈 Pods monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/dRAC0MRIz/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod)
 - [📈 Workloads monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/m_UCAGgIz/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment)
 - [🐘 CNPG some-cnpg-cluster](https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All)
</details>
"
`;

exports[`should render markdown without ingress 1`] = `
"


<details>
  <summary>Docker images</summary>

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

</details>

<details>
  <summary>Debug</summary>

 - [📕 Loki logs for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D)
 - [📈 Pods monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/dRAC0MRIz/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod)
 - [📈 Workloads monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/m_UCAGgIz/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment)

</details>
"
`;

exports[`should render markdown without redirects 1`] = `
"
<details>
  <summary>Ingresses</summary>

 - 🚀 [https://backend-app-preprod.dev.company.com/](https://backend-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/](https://version1-app-preprod.dev.company.com/)
 - 🚀 [https://version1-app-preprod.dev.company.com/something](https://version1-app-preprod.dev.company.com/something)
 - 🚀 [https://version2-app-preprod.dev.company.com/](https://version2-app-preprod.dev.company.com/)

</details>

<details>
  <summary>Docker images</summary>

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

</details>

<details>
  <summary>Debug</summary>

 - [📕 Loki logs for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D)
 - [📈 Pods monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/dRAC0MRIz/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod)
 - [📈 Workloads monitoring for namespace app-preprod](https://grafana.fabrique.social.gouv.fr/d/m_UCAGgIz/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment)
 - [🐘 CNPG some-cnpg-cluster](https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All)
</details>
"
`;
